### ISSUE TITLE
Simplify compute tags

### ISSUE THREAD

**Issue by nilsvu** [sxs-collaboration/spectre#4650]
It would be nice to reduce the amount of boilerplate code in compute tags. @kidder Can we remove the `return_type` alias and just use the `type` of the parent simple tag? Also, can we remove `base`? Or, can we write `MyComputeTag : db::ComputeTag<MySimpleTag>` to inherit those things from the simple tag?

**Comment by kidder** [comment]
yeah, I have a few things like this on my to-do list to clean up for DataBox...
